components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/util/MicroIntegratorBaseUtils.java

```diff
@@ -328,6 +333,21 @@ public static String getDSSJsonFormatter() throws IOException, XMLStreamExceptio
         }
     }
 
+    public static synchronized boolean isHotDeploymentEnabled() {
+        if (hotDeployment == null) {
+            try {
+                String hotDeploymentParam = getPropertyFromAxisConf(
+                        org.wso2.micro.integrator.core.Constants.HOT_DEPLOYMENT);
+                hotDeployment = JavaUtils.isTrue(hotDeploymentParam, false);
+            } catch (IOException | XMLStreamException e) {
+                log.error("Error while reading the " + org.wso2.micro.integrator.core.Constants.HOT_DEPLOYMENT
+                                  + " parameter from axis2.xml.", e);
+                return false;
```

Comment:
Shall we set hotDeployment to false and return here? WDYT?

Reply:
Set hotDeployment to false permanently when an error occurs while reading the axis2.xml for the first time.